### PR TITLE
Add docs for TS `optimizeConstEnums` option

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
Docs for https://github.com/babel/babel/pull/13324

### CODE REVIEW

File: docs/plugin-transform-typescript.md [hunk 1]
```diff
@@ -122,6 +122,48 @@ class A {
...
+  Animals[Animals["Fish"] = 0] = "Fish";
+})(Animals || (Animals = {}));
+
+console.log(Animals.Fish);
+
+// `optimizeConstEnums` output
+console.log(0);
+```
+
+This option differs from TypeScript's `--isolatedModules` behavior, which ignores the `const` modifiers compiling them as normal enums, and aligns Babel's behavior with the TypeScript default behavior.
```
Comment (existentialism) [inline docs/plugin-transform-typescript.md:152]:
```suggestion
This option differs from TypeScript's `--isolatedModules` behavior, which ignores the `const` modifier and compiles them as normal enums, and aligns Babel's behavior with TypeScript's default behavior.
```

File: docs/preset-typescript.md [hunk 2]
```diff
@@ -114,4 +114,46 @@ Added in: `v7.9.0`
...
+  Animals[Animals["Fish"] = 0] = "Fish";
+})(Animals || (Animals = {}));
+
+console.log(Animals.Fish);
+
+// `optimizeConstEnums` output
+console.log(0);
+```
+
+This option differs from TypeScript's `--isolatedModules` behavior, which ignores the `const` modifiers compiling them as normal enums, and aligns Babel's behavior with the TypeScript default behavior.
```
Comment (existentialism) [inline docs/preset-typescript.md:144]:
```suggestion
This option differs from TypeScript's `--isolatedModules` behavior, which ignores the `const` modifier and compiles them as normal enums, and aligns Babel's behavior with TypeScript's default behavior.
```

**Review by existentialism — APPROVED** [review]

**Comment by hanzooo** [comment]
Seem that the [caveats part 2](https://github.com/babel/website/blob/6ca31fdba9c305d6119af956022f066f53b5c165/docs/plugin-transform-typescript.md#caveats) can be updated too.

**Comment by nicolo-ribaudo** [comment]
Uh thanks for the heads up! Would you like to open a PR?

**Comment by hanzooo** [comment]
> Uh thanks for the heads up! Would you like to open a PR?

sure, I just open a PR

**Pull request merged**